performance-analysis/database.py

```diff
@@ -0,0 +1,308 @@
+import sqlite3
+from typing import Iterable
+
+
+_INTEGER = 'INTEGER'
+_PRIMARY_KEY = 'PRIMARY KEY'
+_REAL = 'REAL'
+_TEXT = 'TEXT'
+
+IdMap = dict[str, int | None]
+
+
+class _Context(object):
+    _connection: sqlite3.Connection
+    _cursor: sqlite3.Cursor
+
+    def __init__(self, connection: sqlite3.Connection) -> None:
+        self._connection = connection
+        self._cursor = self._connection.cursor()
+
+    def execute(self, statement: str) -> None:
+        self._execute(statement)
+
+    def insert(self, table_name: str, fields: list[str], values: list) -> None:
+        statement = f'''INSERT INTO {table_name} ({", ".join(fields)})
+                        VALUES ({", ".join(["?"] * len(fields))})'''
+
+        #print(' '.join([line for line in [line.strip() for line in statement.splitlines()] if line != '']))
```

Comment:
:stare:

Reply:
I _suppose_ I could remove the debug logging I accidentally left in there.